inference-engine/src/inference_engine/include/openvino/runtime/async_infer_queue.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief A header file that provides wrapper classes for infer requests and callbacks.
+ *
+ * @file openvino/runtime/async_infer_queue.hpp
+ */
+#pragma once
+
+#include <condition_variable>
+#include <map>
+#include <mutex>
+#include <queue>
+#include <string>
+#include <vector>
+
+#include "openvino/runtime/common.hpp"
+#include "openvino/runtime/executable_network.hpp"
+#include "openvino/runtime/infer_request.hpp"
+#include "openvino/runtime/tensor.hpp"
+
+namespace ov {
+namespace runtime {
+
+/**
+ * @brief This is an interface of asynchronous infer request
+ *
+ * It can throw exceptions safely for the application, where it is properly handled.
+ * 
+ * @note AsyncInferQueue provides simple and unifed interface for multiple asynchronous
+ * calls to OpenVINO's InferRequests.
+ */
+class OPENVINO_RUNTIME_API AsyncInferQueue {
+public:
+    /**
+     * @brief Constructor
+     */
+    AsyncInferQueue(ov::runtime::ExecutableNetwork net, size_t jobs = 0);
+
+    ~AsyncInferQueue() {
+        _requests.clear();
+    }
+
+    /**
+     * @brief Starts asynchronous inference on next avaiable job.
+     */
+    void start_async();
+
+    /**
+     * @brief Starts asynchronous inference with given input data.
+     *
+     * @param inputs Map containing integers (index) and Tensor pairs.
+     */
+    void start_async(std::map<size_t, ov::runtime::Tensor> inputs);
+
+    /**
+     * @brief Starts asynchronous inference with given input data.
+     *
+     * @param inputs Map containing strings (names) and Tensor pairs.
+     */
+    void start_async(std::map<std::string, ov::runtime::Tensor> inputs);
+
+    /**
+     * @brief Waits for any request to become avaiable.
```

Comment:
I suggest adding an explicit mention that this is a blocking call. Consumers of this class should know that without looking at the implementation.